File: lib/src/data/models/product_image.dart
class ProductImage {
  final int imageId;    // PK
  final int productId;  // FK
  final String imageUrl;
  final int sortOrder;

  ProductImage({
    required this.imageId,
    required this.productId,
    required this.imageUrl,
    required this.sortOrder,
  });
}

File: src/main/java/com/example/sneakerx/entities/ProductImage.java
package com.example.sneakerx.entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "product_images")
@Builder
public class ProductImage {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer imageId;

    private String imageUrl;
    private Integer sortOrder;

    @ManyToOne
    @JoinColumn(name = "product_id")
    @JsonIgnore // <--- THE FIX: Stops Jackson from serializing the Cart back to the parent
    @ToString.Exclude // <--- Stop Lombok from printing it
    @EqualsAndHashCode.Exclude
    private Product product;

}
